[Bug]: Error when "copy field" in Calculator transform if lazy conversion field

### Apache Hop version?

SNAPSHOT-20230109

### Java version?

openjdk version "11.0.9.1" 2020-11-04 OpenJDK Runtime Environment (build 11.0.9.1+1-post-Uos-1deb10u2) OpenJDK 64-Bit Server VM (build 11.0.9.1+1-post-Uos-1deb10u2, mixed mode, sharing)

### Operating system

Linux

### What happened?

"Copy field" in "Calculator" transform raise error in field from csv text input with lazy conversion enabled, like formula did in https://issues.apache.org/jira/browse/HOP-4587.

Sample pipeline with data:
[copyField.zip](https://github.com/apache/hop/files/10382463/copyField.zip)

Log:
```
2023/01/10 12:58:29 - Hop - Pipeline opened.
2023/01/10 12:58:29 - Hop - Launching pipeline [copyField]...
2023/01/10 12:58:29 - Hop - Started the pipeline execution.
2023/01/10 12:58:29 - copyField - Executing this pipeline using the Local Pipeline Engine with run configuration 'local'
2023/01/10 12:58:29 - copyField - Execution started for pipeline [copyField]
2023/01/10 12:58:29 - [copyField].asset.csv.0 - Header row skipped in file 'file:///home/enrico/repositories/hop/cdg/etl/nico/asset.csv'
2023/01/10 12:58:29 - [copyField].asset.csv.0 - Finished processing (I=18, O=0, R=0, W=17, U=0, E=0)
2023/01/10 12:58:29 - [copyField].newhostid.0 - Finished processing (I=0, O=0, R=17, W=17, U=0, E=0)
2023/01/10 12:58:29 - copyField - Pipeline duration : 0.164 seconds [  0.164" ]
2023/01/10 12:58:29 - copyField - ERROR: Error handling writing final pipeline state to location (non-fatal)
2023/01/10 12:58:29 - copyField - ERROR: org.apache.hop.core.exception.HopException: 
2023/01/10 12:58:29 - copyField - Error getting execution from Neo4j
2023/01/10 12:58:29 - copyField - org.apache.hop.core.exception.HopValueException: 
2023/01/10 12:58:29 - copyField - Unexpected conversion error while converting value [newhostid Integer] to an Integer
2023/01/10 12:58:29 - copyField - class [B cannot be cast to class java.lang.Long ([B and java.lang.Long are in module java.base of loader 'bootstrap')
2023/01/10 12:58:29 - copyField - 
2023/01/10 12:58:29 - copyField - 
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.neo4j.execution.NeoExecutionInfoLocation.registerData(NeoExecutionInfoLocation.java:877)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.pipeline.engines.local.LocalPipelineEngine.stopTransformExecutionInfoTimer(LocalPipelineEngine.java:486)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.pipeline.engines.local.LocalPipelineEngine.pipelineCompleted(LocalPipelineEngine.java:465)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.pipeline.Pipeline.firePipelineExecutionFinishedListeners(Pipeline.java:1343)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.pipeline.Pipeline.lambda$startThreads$0(Pipeline.java:1142)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.pipeline.transform.BaseTransform.fireTransformFinishedListeners(BaseTransform.java:2779)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.pipeline.transform.BaseTransform.markStop(BaseTransform.java:2768)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.pipeline.transform.RunThread.run(RunThread.java:142)
2023/01/10 12:58:29 - copyField - 	at java.base/java.lang.Thread.run(Thread.java:834)
2023/01/10 12:58:29 - copyField - Caused by: java.lang.RuntimeException: org.apache.hop.core.exception.HopValueException: 
2023/01/10 12:58:29 - copyField - Unexpected conversion error while converting value [newhostid Integer] to an Integer
2023/01/10 12:58:29 - copyField - class [B cannot be cast to class java.lang.Long ([B and java.lang.Long are in module java.base of loader 'bootstrap')
2023/01/10 12:58:29 - copyField - 
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.neo4j.execution.NeoExecutionInfoLocation.saveNeo4jRowsAndMeta(NeoExecutionInfoLocation.java:1115)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.neo4j.execution.NeoExecutionInfoLocation.registerNeo4jData(NeoExecutionInfoLocation.java:944)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.neo4j.execution.NeoExecutionInfoLocation.lambda$registerData$11(NeoExecutionInfoLocation.java:875)
2023/01/10 12:58:29 - copyField - 	at org.neo4j.driver.internal.InternalSession.lambda$transaction$4(InternalSession.java:150)
2023/01/10 12:58:29 - copyField - 	at org.neo4j.driver.internal.retry.ExponentialBackoffRetryLogic.retry(ExponentialBackoffRetryLogic.java:103)
2023/01/10 12:58:29 - copyField - 	at org.neo4j.driver.internal.InternalSession.transaction(InternalSession.java:146)
2023/01/10 12:58:29 - copyField - 	at org.neo4j.driver.internal.InternalSession.writeTransaction(InternalSession.java:124)
2023/01/10 12:58:29 - copyField - 	at org.neo4j.driver.internal.InternalSession.writeTransaction(InternalSession.java:118)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.neo4j.execution.NeoExecutionInfoLocation.registerData(NeoExecutionInfoLocation.java:875)
2023/01/10 12:58:29 - copyField - 	... 8 more
2023/01/10 12:58:29 - copyField - Caused by: org.apache.hop.core.exception.HopValueException: 
2023/01/10 12:58:29 - copyField - Unexpected conversion error while converting value [newhostid Integer] to an Integer
2023/01/10 12:58:29 - copyField - class [B cannot be cast to class java.lang.Long ([B and java.lang.Long are in module java.base of loader 'bootstrap')
2023/01/10 12:58:29 - copyField - 
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.core.row.value.ValueMetaBase.getInteger(ValueMetaBase.java:2368)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.core.row.value.ValueMetaInteger.getNativeDataType(ValueMetaInteger.java:40)
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.neo4j.execution.NeoExecutionInfoLocation.saveNeo4jRowsAndMeta(NeoExecutionInfoLocation.java:1081)
2023/01/10 12:58:29 - copyField - 	... 16 more
2023/01/10 12:58:29 - copyField - Caused by: java.lang.ClassCastException: class [B cannot be cast to class java.lang.Long ([B and java.lang.Long are in module java.base of loader 'bootstrap')
2023/01/10 12:58:29 - copyField - 	at org.apache.hop.core.row.value.ValueMetaBase.getInteger(ValueMetaBase.java:2281)
2023/01/10 12:58:29 - copyField - 	... 18 more
2023/01/10 12:58:29 - copyField - Execution finished on a local pipeline engine with run configuration 'local'
```

### Issue Priority

Priority: 3

### Issue Component

Component: Transforms

--- a/plugins/transforms/calculator/src/main/java/org/apache/hop/pipeline/transforms/calculator/calculations/CopyField.java
+++ b/plugins/transforms/calculator/src/main/java/org/apache/hop/pipeline/transforms/calculator/calculations/CopyField.java
@@ -17,6 +17,7 @@
 
 package org.apache.hop.pipeline.transforms.calculator.calculations;
 
+import org.apache.hop.core.exception.HopValueException;
 import org.apache.hop.core.row.IValueMeta;
 import org.apache.hop.i18n.BaseMessages;
 import org.apache.hop.pipeline.transforms.calculator.CalculationInput;
@@ -43,7 +44,12 @@ public class CopyField implements ICalculation {
   }
 
   @Override
-  public CalculationOutput calculate(CalculationInput in) {
+  public CalculationOutput calculate(CalculationInput in) throws HopValueException {
+    if (in.dataA instanceof byte[] && in.metaA != null) {
+      Object convertedData = in.metaA.convertBinaryStringToNativeType((byte[]) in.dataA);
+      return new CalculationOutput(in.metaA.getType(), convertedData);
+    }
+
     return new CalculationOutput(ICalculation.getResultType(in.metaA), in.dataA);
   }
 }

--- a/plugins/transforms/calculator/src/test/java/org/apache/hop/pipeline/transforms/calculator/CalculatorMetaTest.java
+++ b/plugins/transforms/calculator/src/test/java/org/apache/hop/pipeline/transforms/calculator/CalculatorMetaTest.java
@@ -18,6 +18,7 @@ package org.apache.hop.pipeline.transforms.calculator;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.util.Random;
 import java.util.UUID;
@@ -53,7 +54,7 @@ class CalculatorMetaTest {
     for (int i = 0; i < meta.getFunctions().size(); i++) {
       CalculatorMetaFunction function = meta.getFunctions().get(i);
       CalculatorMetaFunction function2 = meta2.getFunctions().get(i);
-      assertEquals(function, function2);
+      assertTrue(function.equals(function2));
     }
   }
 

